src/geowombat/core/io.py

```diff
@@ -950,126 +962,125 @@ def to_raster(
         tqdm_kwargs = {}
 
     # Force tiled outputs with no file sharing
-    kwargs['sharing'] = False
+    kwargs["sharing"] = False
 
     if data.gw.tiled:
-        kwargs['tiled'] = True
+        kwargs["tiled"] = True
 
-    if 'compress' in kwargs:
+    if "compress" in kwargs:
         # boolean True or '<>'
-        if kwargs['compress']:
+        if kwargs["compress"]:
             if (
-                isinstance(kwargs['compress'], str)
-                and kwargs['compress'].lower() == 'none'
+                isinstance(kwargs["compress"], str)
+                and kwargs["compress"].lower() == "none"
             ):
                 compress = False
             else:
-                if 'num_threads' in kwargs:
+                if "num_threads" in kwargs:
                     compress = False
                 else:
                     compress = True
 
                 if compress:
                     # Store the compression type because
                     #   it is removed in concurrent writing
-                    compress_type = kwargs['compress']
-                    del kwargs['compress']
+                    compress_type = kwargs["compress"]
+                    del kwargs["compress"]
 
         else:
             compress = False
 
     elif isinstance(data.gw.compress, str) and (
-        data.gw.compress.lower() in ['lzw', 'deflate']
+        data.gw.compress.lower() in ["lzw", "deflate"]
     ):
         compress = True
         compress_type = data.gw.compress
 
     else:
         compress = False
 
-    if 'nodata' not in kwargs:
-        if isinstance(data.gw.nodata, int) or isinstance(
-            data.gw.nodata, float
-        ):
-            kwargs['nodata'] = data.gw.nodata
+    if kwargs.get("nodata") is None:
```

Comment:
@mmann1123 this is the fix for ensuring the `nodata` value gets saved. Everything else in this file is auto-formatting.